### PR TITLE
[JENKINS-73031] Print hint that a `static` `interface` method may be to blame

### PR DESCRIPTION
**Pull request by jglick**
https://issues.jenkins.io/browse/JENKINS-73031?focusedId=449518&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-449518

### CODE REVIEW

File: plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java [hunk 1]
```diff
@@ -435,6 +435,16 @@ private boolean run() {
                     if (fn != null) {
                         t.head.get().addAction(new ErrorAction(error));
                     }
+                    if (error instanceof VerifyError) {
+                        var msg = error.getMessage();
+                        if (msg != null && msg.contains("Illegal type in constant pool")) {
+                            try {
+                                execution.getOwner().getListener().getLogger().println("May be JENKINS-73031: static interface methods not yet supported in Groovy");
```
Comment (dwnusbaum) [inline plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java:442]: I would maybe rephrase this to indicate that it is more of a Jenkins problem than a Groovy problem:

```suggestion
                                execution.getOwner().getListener().getLogger().println("May be JENKINS-73031: static interface methods are not currently supported");
```

**Review by dwnusbaum — APPROVED** [review]

File: plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java (hunk 1 above)
Comment (jglick) [reply plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java:442]: Good point. Or what about
```suggestion
                                execution.getOwner().getListener().getLogger().println("May be JENKINS-73031: calling static interface methods from Groovy is not currently supported by Jenkins");
```
?

Comment (dwnusbaum) [reply plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java:442]: Yes that looks good to me.

**Review by dwnusbaum — APPROVED** [review]
